src/solidOpts.ts

```diff
@@ -21,7 +24,24 @@ let elementDeleteQueue: ElementNode[] = [];
 function flushDeleteQueue(): void {
   for (let el of elementDeleteQueue) {
     if (Number(el._queueDelete) < 0) {
-      el.destroy();
+      if (el.preserve) {
+        el.alpha = 0;
+        if (!el._hasCleanup) {
+          el._hasCleanup = true;
+          const removeIt = el.destroy.bind(el);
+          s.runWithOwner(el.preserve, () => {
+            s.onCleanup(() => {
+              removeIt();
+              console.log('removed');
+            });
+          });
+        }
+      } else {
+        el.destroy();
+      }
+    } else {
+      // inserted back, make it visible again
+      el.alpha = 1;
```

Comment:
This assumes the element doesn't have `opacity={0.6}`
Why change the opacity anyway?

Reply:
This is the simplist way to hide an element and bring it back. I agree changing the node values might not be ideal.

Reply:
Yeah adding `onInsert` and `onRemove` is generally what I was thinking about. Which could just replace `onCreate` and `onDestroy` for animations.